opt_adapt/opt.py

```diff
@@ -184,7 +190,120 @@ def _gradient_descent(it, forward_run, m, params, u, u_, dJ_):
 
     # Take a step downhill
     u -= lr * dJ
-    yield {"lr": lr, "u+": u, "u-": u_, "dJ-": dJ_}
+    yield {"lr": lr, "u+": u}
+
+
+def _adam(it, forward_run, m, params, u, a_, b_, beta_1=0.9, beta_2=0.999, epsilon=1e-8):
```

Comment:
I don't think the user ever gets to modify the `beta_1`, `beta_2` or `epsilon` parameters. Perhaps you could put them in the parameter class?